src/Compilers/CSharp/Portable/Emitter/Model/MethodSymbolAdapter.cs

```diff
@@ -164,7 +164,11 @@ private Cci.IMethodDefinition ResolvedMethodImpl(EmitContext context)
             if (AdaptedMethodSymbol.IsDefinition && // can't be generic instantiation
                 AdaptedMethodSymbol.ContainingModule == moduleBeingBuilt.SourceModule) // must be declared in the module we are building
             {
-                Debug.Assert((object)AdaptedMethodSymbol.PartialDefinitionPart == null); // must be definition
+                if (AdaptedMethodSymbol.PartialDefinitionPart is { } definition)
+                {
+                    return definition.GetCciAdapter().ResolvedMethodImpl(context);
```

Comment:
This change was needed to avoid crash in the sequence point test (added in the same commit). Similar test would also crash for partial methods/properties. Basically the VerifyMethodBody calls GetResolvedMethod on the implementation Cci symbol - that probably doesn't happen normally during emit, that's why the assert was fine for normal scenarios.

Reply:
This change doesn't look right. I think the API is not supposed to return a different instance, it should either return an original instance, or null. Basically, for a pair of partial declarations, we should never "create" two different adapters, translation layer should make sure of that. Symbols should never be casted to CCI interfaces, they should be translated.

Reply:
If I remember correctly, PEModuleBuilder performs the translation. There are APIs called "Translate..."

Reply:
Okay, I think I can skip the failing test for now and look at this separately.